ledger/src/blockstore_processor.rs

```diff
@@ -290,6 +292,79 @@ fn execute_batches(
     first_err(&results)
 }
 
+fn execute_batches(
+    bank: &Arc<Bank>,
+    batches: &[TransactionBatch],
+    entry_callback: Option<&ProcessCallback>,
+    transaction_status_sender: Option<&TransactionStatusSender>,
+    replay_vote_sender: Option<&ReplayVoteSender>,
+    timings: &mut ExecuteTimings,
+    cost_capacity_meter: Arc<RwLock<BlockCostCapacityMeter>>,
+) -> Result<()> {
+    let lock_results = batches
+        .iter()
+        .flat_map(|batch| batch.lock_results().clone())
+        .collect::<Vec<_>>();
+    let sanitized_txs = batches
+        .iter()
+        .flat_map(|batch| batch.sanitized_transactions().to_vec())
+        .collect::<Vec<_>>();
+
+    let cost_model = CostModel::new();
+    let mut minimal_tx_cost = u64::MAX;
+    let total_cost: u64 = sanitized_txs
+        .iter()
+        .map(|tx| {
+            let cost = cost_model.calculate_cost(tx).sum();
```

Comment:
I don't know how expensive 'calculate_cost' is. If it is expensive at all, you could save the result per tx in a vec in the first pass. Line 333 below could lookup the cost you save away. I know we've talked about loading the account and determining the data size to use as part of the cost. That could make calculate_cost more expensive.

Reply:
That's a good point. I was not sure how expensive is this operation. Agree that we can call it once and save the results in a vec. I'll rework the code.